server/controllers/request.js

```diff
@@ -1,25 +1,58 @@
 const Request = require("../models/Request");
 const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
 
 // @route GET /requests
 // @desc list of requests for logged in user
 // @access Private
 exports.getRequests = asyncHandler(async (req, res, next) => {
-  const user = req.body.userId;
+  const userId = req.body.userId;
 
-  let requests;
-  if (user) {
-    requests = await Request.find({
-      $or: [{ userId: user }, { sitterId: user }],
-    });
-  }
+  const requests = await Request.find({
+    $or: [{ userId: userId }, { sitterId: userId }],
+  });
 
-  if (!user) {
-    res.status(404);
+  if (!requests) {
+    res.status(400);
     throw new Error("No quest found.");
   }
 
-  res.status(200).json({ requests: requests });
+  const makeRequestObj = async (id, request) => {
+    let user = await User.findOne({ _id: id });
+    let requestObj = {
+      id: request._id,
+      username: user.username,
+      url: "https://robohash.org/mockLoggedInUser3@gmail.com.png",
+      date: request.start.toGMTString().slice(5, 16),
+      time: `${request.start.getHours()}:${request.start.getMinutes()} - ${request.end.getHours()}:${request.end.getMinutes()} ${request.end
+        .toLocaleTimeString()
+        .slice(-2)}`,
+      start: request.start,
+      end: request.end,
+      timeZone: request.timeZone,
+      accepted: request.accepted,
+      declined: request.declined,
+      paid: request.paid,
+      requestDate: request.requestDate,
+    };
+    return requestObj;
+  };
```

Comment:
This is interesting but not necessary, since the user id is in the request object, we should be able to populate if we reference the user on the request model. Once that is done we can refactor to populate the user information to the request object and we don't really need to recreate the request model structure as MongoDB will already pass in the structure.

Reply:
tl;dr

Once you have added reference to the users in the Request model, just search for the request and populate the user and sitter on that model so we have everything we need and pass that directly to the response (we don't need makeRequestObj anymore. 😄